src/firmware/app/control/trajectory_planner_test.cpp

```diff
@@ -1288,50 +768,1631 @@ TEST_F(TrajectoryPlannerTest, velocity_trajectory_curved_path_low_acceleration)
+    EXPECT_NEAR(trajectory.trajectory_elements[2].linear_speed, sqrt(3), 0.0001);
+    EXPECT_NEAR(trajectory.trajectory_elements[3].linear_speed, 1, 0.0001);
+}
+
+TEST_F(TrajectoryPlannerTest,
+       test_backwards_continuity_variable_segment_length_continuous)
+{
+    float max_allowable_speed_profile[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS];
+
+    const unsigned int num_segments = 4;
+
+
+    TrajectorySegment_t segment_lengths[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS];
+    segment_lengths[0].linear_segment_length = 1.0;
+    segment_lengths[1].linear_segment_length = 2.0;
+    segment_lengths[2].linear_segment_length = 3.0;
+
+
+    FirmwareRobotPathParameters_t path_parameters;
+    path_parameters.max_allowable_angular_acceleration = 1;
+    path_parameters.max_allowable_angular_speed        = 1;
+    path_parameters.max_allowable_linear_acceleration  = 1;
+    path_parameters.max_allowable_linear_speed         = 1;
+    path_parameters.num_segments                       = num_segments;
+    path_parameters.initial_linear_speed               = sqrt(7);
+    path_parameters.final_linear_speed                 = 1;
+
+    float speed_profile[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS];
+
+
+    PositionTrajectory_t trajectory;
+    PositionTrajectoryElement_t trajectory_elements[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS];
+    trajectory.path_parameters     = path_parameters;
+    trajectory.trajectory_elements = trajectory_elements;
+
+    trajectory.trajectory_elements[0].linear_speed = 4;
+    trajectory.trajectory_elements[1].linear_speed = 3;
+    trajectory.trajectory_elements[2].linear_speed = 10;
+    trajectory.trajectory_elements[3].linear_speed = 1;
+
+    TrajectoryPlannerGenerationStatus_t status =
+        app_trajectory_planner_modifyTrajectoryToBackwardsContinuous(&trajectory,
+                                                                     segment_lengths);
+    EXPECT_EQ(status, OK);
+
+    EXPECT_NEAR(trajectory.trajectory_elements[0].linear_speed, sqrt(11), 0.0001);
+    EXPECT_NEAR(trajectory.trajectory_elements[1].linear_speed, 3, 0.0001);
+    EXPECT_NEAR(trajectory.trajectory_elements[2].linear_speed, sqrt(7), 0.0001);
+    EXPECT_NEAR(trajectory.trajectory_elements[3].linear_speed, 1, 0.0001);
+}
+
+TEST_F(TrajectoryPlannerTest,
+       test_backwards_continuity_variable_segment_length_initial_velocity_too_high)
+{
+    // Create an 'initial velocity too high' state by requesting an initial velocity that
+    // is much greater than physically possible given the path length and acceleration
+    float max_allowable_speed_profile[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS];
+
+    const unsigned int num_segments = 4;
+
+
+    TrajectorySegment_t segment_lengths[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS];
+    segment_lengths[0].linear_segment_length = 1.0;
+    segment_lengths[1].linear_segment_length = 2.0;
+    segment_lengths[2].linear_segment_length = 3.0;
+
+
```

Comment:
nit; extra blanks line here and above

Reply:
looks like this was copied elsewhere. Not super important, but would appreciate it if you could do a quick pass to remove excess blank lines, unless they're intentional.

Reply:
Did a pass of the tests and cleaned up whitespace 👍